fs/inode/fs_files.c

```diff
@@ -858,6 +1027,263 @@ int fs_putfilep(FAR struct file *filep)
 }
 #endif
 
+/****************************************************************************
+ * Name: nx_fcntl
+ *
+ * Description:
+ *   nx_fcntl() will perform the operation specified by 'cmd' on an open
+ *   file.
+ *
+ * Input Parameters:
+ *   fd  - File descriptor of the open file
+ *   cmd - Identifies the operation to be performed.  Command specific
+ *         arguments may follow.
+ *   arg - The argument.
+ *
+ * Returned Value:
+ *   A non-negative value is returned on success; a negated errno value is
+ *   returned on any failure.
+ *
+ ****************************************************************************/
+
+int nx_fcntl(int fd, int cmd, unsigned long arg)
```

Comment:
> where do we implement nx_fcntl

@xiaoxiang781216 it's here.

Reply:
should we keep it in fs_fcntl.c

Reply:
I'm trying to hide fl_lock semantics inside fs_files.c. We could make getters/setters for all of the file descriptor flags (closeonexec, fdsan, fdcheck) but this would generate a lot of code.